hack/jenkins/e2e.sh

```diff
@@ -80,8 +103,14 @@ fi
 # Have cmd/e2e run by goe2e.sh generate JUnit report in ${WORKSPACE}/junit*.xml
 export E2E_REPORT_DIR=${WORKSPACE}
 
+function teardown() {
+  echo "Cleaning up with e2e --down"
+  go run ./hack/e2e.go ${E2E_OPT} -v --down
+}
```

Comment:
Ack, sorry, I literally just forced another rebase on you. Well. If #4350 goes in.

I opted for the approach that was simpler than the trap, but I'm willing to take the trap, too. I was actually about to take the _trap_error approach here, honestly: http://unix.stackexchange.com/questions/125607/proper-usage-of-exit-err-traps-in-bash-scripts .. because I had a long day once where something really hokey was going on with errexit.

I have a mild preference for #4350 going in first and then we can let that settle in just to check. I have yet to setup a test branch for Jenkins, which I really ought to do.